Compiler warnings in macOS native build

macOS 10.15.7 (see https://github.com/bitshares/bitshares-core/runs/3627081567)


> 2021-09-16T23:53:35.9359200Z /Users/runner/work/bitshares-core/bitshares-core/libraries/fc/include/fc/static_variant.hpp:91:13: warning: destructor called on non-final 'graphene::protocol::transfer_operation' that has virtual functions but non-virtual destructor [-Wdelete-non-abstract-non-virtual-dtor]
2021-09-16T23:53:35.9361690Z             reinterpret_cast<T*>(data)->~T();
2021-09-16T23:53:35.9362450Z             ^

By the way, the same warnings appear in Linux cross-build too (see #2340).

--- a/libraries/protocol/include/graphene/protocol/base.hpp
+++ b/libraries/protocol/include/graphene/protocol/base.hpp
@@ -123,6 +123,7 @@ namespace graphene { namespace protocol {
 
    struct base_operation
    {
+      virtual ~base_operation() = default;
       template<typename T>
       share_type calculate_fee(const T& params)const
       {

